src/main/user-impl/java/com/mysql/cj/jdbc/ha/ConnectionProxy.java

```diff
@@ -196,14 +203,10 @@ protected void initLogger(ConnectionUrl connUrl) {
     }
   }
 
-  protected void initProxy() {
-    this.initProxy(ConnectionPluginManager::new);
-  }
-
   protected void initSettings(ConnectionUrl connectionUrl) throws SQLException {
-    final JdbcPropertySetImpl connProps = new JdbcPropertySetImpl();
     try {
-      connProps.initializeProperties(connectionUrl.getMainHost().exposeAsProperties());
+      final JdbcPropertySetImpl connProps = new JdbcPropertySetImpl();
+      connProps.initializeProperties(connectionUrl.getConnectionArgumentsAsProperties());
```

Comment:
Why are we using `connectionUrl.getConnectionArgumentsAsProperties()`? Using this would lose some config, such as database name.